### PR TITLE
Extend signup timeout to 2 hours from 30 minutes

### PR DESCRIPTION
**Pull request by jim**
# What it does

We had an instance of a member taking longer than 30 minutes to complete the signup steps, and it timed out just as they were paying for their membership. You can see [the sample of this occurrence in AppSignal](https://appsignal.com/chicago-tool-library/sites/60596f9214ad662e17191689/performance/incidents/59/samples/60596f9214ad662e17191689-158228531489123241821718233200?selectedTime=2024-06-12T23%3A05%3A00.000Z&activeFilter=date). This meant that the callback from Square was rejected and their membership payment wasn't recorded correctly in our system. The logic to calculate if the time had passed was also hard to follow as it was previously implemented.

This change gives people 2 hours to get through signup. The original time was chosen arbitrarily, and in hindsight I can't say why it needs to be so short. I'm open to making it even longer; the main risk I can think of is on shared computers where folks might enter some of their info, walk away, and leave the window open for someone else to complete.

The other part of a complete solution to this problem is to implement web hooks for payments, which it captured by #1556.

### CODE REVIEW

File: app/controllers/signup/base_controller.rb [hunk 1]
```diff
@@ -16,7 +18,7 @@ def is_membership_enabled?
     private
 
     def load_member
-      if session[:member_id] && session[:timeout] && session[:timeout] > Time.current - 15.minutes
```
Comment (jim) [inline app/controllers/signup/base_controller.rb:19]: This logic was really hard for me to wrap my head around. Now, `session[:timeout]` contains a time, after which the current session is considered invalid.

**Review by crismali — APPROVED** [review]
Looks great to me!
